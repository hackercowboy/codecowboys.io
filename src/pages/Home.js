import React from 'react';
import PropTypes from 'prop-types';
import { Parallax } from 'react-parallax';
import { injectIntl } from 'react-intl';

import Main from '../components/Main';
import Header from '../components/Header';

import Navigation from '../components/Navigation';
import NavigationItem from '../components/NavigationItem';

import ApplicationCarousel from '../components/ApplicationCarousel';
import ApplicationHero from '../components/ApplicationHero';
import Application from '../components/Application';
import Section from '../components/Section';
import ContactForm from '../components/ContactForm';
import About from '../components/About';

import ScrollButton from '../components/ScrollButton';

const letterCarouselImage = require('../assets/images/letter_carousel.png');
const heroBackground = require('../assets/images/bg-6.jpg');
const letterIcon = require('../assets/images/letter_icon.svg');
const aboutBackground = require('../assets/images/bg-5.jpg');

function Home({ intl }) {
  return (
    <Main className="presentation">
      <Header>
        <Navigation>
          <NavigationItem target="apps" offset={-76}>{intl.formatMessage({ id: 'home.apps_title' })}</NavigationItem>
          <NavigationItem target="about" offset={-5}>{intl.formatMessage({ id: 'about.title' })}</NavigationItem>
          <NavigationItem target="contact" offset={-76}>{intl.formatMessage({ id: 'contact.title' })}</NavigationItem>
        </Navigation>
      </Header>
      <Parallax
        bgImage={heroBackground}
        strength={500}
      >
        <ApplicationCarousel>
          <ApplicationHero
            image={letterCarouselImage}
            title={intl.formatMessage({ id: 'letter.title' })}
            link={intl.formatMessage({ id: 'letter.link' })}
            appStoreLink="http://itunes.apple.com/app/letter/id498506154"
            description={intl.formatMessage({ id: 'letter.teaser' })}
          />
        </ApplicationCarousel>
        <ScrollButton target="apps" />
      </Parallax>
      <Section
        id="apps"
        title={intl.formatMessage({ id: 'home.apps_title' })}
        subtitle={intl.formatMessage({ id: 'home.apps_subtitle' })}
        offset={-76}
      >
        <Application
          title={intl.formatMessage({ id: 'letter.title' })}
          teaser={intl.formatMessage({ id: 'letter.teaser' })}
          link={intl.formatMessage({ id: 'letter.link' })}
          image={letterIcon}
        />
      </Section>
      <Parallax
        bgImage={aboutBackground}
        strength={500}
      >
        <Section
          id="about"
          title={intl.formatMessage({ id: 'about.title' })}
          subtitle={intl.formatMessage({ id: 'about.subtitle' })}
          dark
        >
          <About />
        </Section>
      </Parallax>
      <ContactForm />
    </Main>
  );
}

Home.propTypes = {
  intl: PropTypes.shape({ formatMessage: PropTypes.func.isRequired }).isRequired,
};

export default injectIntl(Home);
