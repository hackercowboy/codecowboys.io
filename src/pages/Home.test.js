import React from 'react';
import { shallow } from 'enzyme';

import Home from './Home';

describe('<Home/>', () => {
  it('should render', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper.shallow().debug()).toMatchSnapshot();
  });
});
