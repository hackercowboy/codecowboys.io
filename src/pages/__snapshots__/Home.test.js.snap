// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> should render 1`] = `
"<Content space={false}>
  <Navigation>
    <NavigationItem target=\\"apps\\" offset={-85} href={[undefined]} active={false} onSetActive={[undefined]}>
      Apps
    </NavigationItem>
    <NavigationItem target=\\"about\\" offset={-5} href={[undefined]} active={false} onSetActive={[undefined]}>
      About
    </NavigationItem>
    <NavigationItem target=\\"contact\\" offset={-85} href={[undefined]} active={false} onSetActive={[undefined]}>
      Contact
    </NavigationItem>
  </Navigation>
  <Parallax bgImage=\\"bg-6.jpg\\" strength={500} bgClassName=\\"react-parallax-bgimage\\" bgImageAlt=\\"\\" className=\\"\\" contentClassName=\\"\\" disabled={false}>
    <ApplicatinCarousel>
      <ApplicationHero image=\\"letter_carousel.png\\" title=\\"Letter\\" link=\\"/en/letter\\" appStoreLink=\\"http://itunes.apple.com/app/letter/id498506154\\" description=\\"With the Letter app you can now write your personal letters on the iPhone, iPad and iPod Touch too, when and where you want.\\" soon={false} />
    </ApplicatinCarousel>
  </Parallax>
  <Section id=\\"apps\\" title=\\"Apps\\" subtitle=\\"A couple of great apps for you!\\" dark={false} light={false} teaser={[undefined]} big={false}>
    <div>
      <div body={true}>
        <h4>
          Letter
        </h4>
        <p>
          With the Letter app you can now write your personal letters on the iPhone, iPad and iPod Touch too, when and where you want.
           
          <a href=\\"/en/letter\\">
            More
            ...
          </a>
        </p>
      </div>
      <div right={true} className=\\"ml-3\\">
        <a href=\\"/en/letter\\">
          <img src=\\"letter_icon.svg\\" style={{...}} alt=\\"Letter\\" />
        </a>
      </div>
    </div>
  </Section>
  <About />
  <ContactForm />
</Main>"
`;
